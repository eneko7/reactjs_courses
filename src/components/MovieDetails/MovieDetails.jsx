import React, { PureComponent } from 'react';
import propTypes from 'prop-types';
import Logo from '../Logo';
import Button from '../Button';
import ImagePoster from '../MoviesGrid/ImagePoster';
import styles from './MovieDetails.scss';
import Loader from '../Loading';
import Genres from '../Genres';

class MovieDetails extends PureComponent {
  componentDidMount() {
    this.getMovie();
  }

  componentWillUnmount() {
    const { fetchMovie } = this.props;
    return fetchMovie(0);
  }

  getMovie = () => {
    const { location: { pathname }, fetchMovie } = this.props;
    return fetchMovie(pathname.substr(7));
  };

  returnMain = () => {
    const { history } = this.props;
    history.push('');
  };

  render() {
    const { movie, fetchSearchMovies } = this.props;
    if (!movie) return <Loader />;
    const { genres } = movie;
    if (genres) {
      fetchSearchMovies(genres.join(', '), 'genres', 'release_date');
    }
    return (
      <div className={styles.movie_details}>
        <div className={styles.top_line}>
          <Logo />
          <Button classButton="details_button" text="search" onClick={this.returnMain} />
        </div>
        <div className={styles.main}>
          <ImagePoster src={movie.poster_path} classPoster="poster" />
          <div className={styles.wrapper_right}>
            <span className={styles.movie_titile}>{movie.title}</span>
            <Genres data={movie.genres} classGenres="genresClass" />
            <div className={styles.wrapper_year_and_time}>
              <span className={styles.year}>
                {new Date(movie.release_date).getFullYear()}
              </span>
              <span className={styles.time}>{movie.runtime ? ` ${movie.runtime}  min` : ''}</span>
            </div>
            <div className={styles.movie_description}>{movie.overview}</div>
          </div>
        </div>
      </div>
    );
  }
}
MovieDetails.defaultProps = {
  movie: null,
};

MovieDetails.propTypes = {
  history: propTypes.objectOf(propTypes.any).isRequired,
  location: propTypes.shape({
    params: propTypes.object,
    pathname: propTypes.string,
    search: propTypes.string,
  }).isRequired,
  movie: propTypes.objectOf(propTypes.any),
  fetchMovie: propTypes.func.isRequired,
  fetchSearchMovies: propTypes.func.isRequired,
};

export default MovieDetails;
