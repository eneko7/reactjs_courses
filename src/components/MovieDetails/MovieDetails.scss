@import "../../assets/constants/constants.scss";

.movie_details {
  height: 63vh;
  background-image: url('../../assets/images/netflix-image.jpg');
  background-position: right top;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
  z-index: 3;
  padding: 1% 15%;
  min-height: 260px;

  .top_line {
    position: relative;
    width: 100%;
    display: flex;
    justify-content: space-between;
    z-index: 3;
    margin-bottom: 20px;
  }

  .main {
    position: relative;
    z-index: 3;
    display: flex;
    justify-content: space-between;

    .wrapper_right {
      padding: 2% 0 0 5%;
      width: 70%;

      .movie_titile {
        color: $roseColor;
        font-family: $fontFamilyBold;
        font-size: 40px;
      }

      .wrapper_year_and_time {
        margin-bottom: 30px;

        .year,
        .time {
          color: #fff;
          font-family: $fontFamilyBold;
          margin-right: 10%;
        }
      }

      .movie_description {
        color: #fff;
        font-family: $fontFamilyRegular;
        font-size: 21px;
        line-height: 32px;
      }
    }
  }
}

.movie_details::after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.6);
  z-index: 2;
}

@media (max-width: $fulhdDesktop) {
  .movie_details {
    padding: 1% $leftPaddingLaptop;
  }
}

@media (max-width: $laptop) {
  .movie_details {
    padding: 1% $leftPaddingTablet;
  }
}

@media (max-width: $tablet) {
  .movie_details {
    padding: 1% 5%;
    height: 32vh;
  }
}
