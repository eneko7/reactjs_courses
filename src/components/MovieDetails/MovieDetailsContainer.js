import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import MovieDetails from './MovieDetails';
import { fetchMovie } from '../../modules/movie/movieActions';
import { fetchSearchMovies } from '../../modules/movies/moviesActions';
import { getMovieState, getIsFetchingMovie, getIsFetchedMovie } from '../../modules/movie/movieSelectors';

const mapStateToProps = state => ({
  movie: getMovieState(state),
  isFetchingMovie: getIsFetchingMovie(state),
  isFetchedMovie: getIsFetchedMovie(state),
});

const mapDispatchToProps = {
  fetchMovie,
  fetchSearchMovies,
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MovieDetails));
