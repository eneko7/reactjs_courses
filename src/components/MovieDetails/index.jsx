import MovieDetails from './MovieDetailsContainer';

export default MovieDetails;
