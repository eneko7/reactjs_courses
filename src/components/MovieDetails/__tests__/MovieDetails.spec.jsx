import React from 'react';
import TestRenderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import MovieDetails from '../MovieDetails';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
jest.mock('../MovieDetailsContainer', () => () => <div>test</div>);
const historyMock = { push: jest.fn() };
const data = {
  movies: [
    {
      id: 412059,
      title: 'Mobile Homes',
      tagline: '',
      vote_average: 0,
      vote_count: 1,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
      overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
      budget: 0,
      revenue: 0,
      genres: [
        'Drama',
      ],
      runtime: 105,
    },
    {
      id: 424585,
      title: 'Mademoiselle Paradis',
      tagline: '',
      vote_average: 6,
      vote_count: 2,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/k93b9hE30fdYG63atluVP5aDm2V.jpg',
      overview: '18th century Vienna. The true story of Maria Theresia von Paradis, a gifted piano player and close friend of Mozart’s, who lost her eye-sight as a child. Desperate to cure their talented daughter, the Paradis entrust Maria to Dr. Mesmer, a forward-thinking-physician who gives her the care and attention that she requires. With the doctor’s innovative techniques of magnetism, Maria slowly recovers her sight. But this miracle comes at a price as the woman progressively starts to lose her gift for music. Faced with a heavy dilemma, Mademoiselle Paradis will have to choose: an ordinary life in the light or an extraordinary life in darkness, as a virtuoso.',
      budget: 0,
      revenue: 0,
      genres: [
        'History',
        'Drama',
      ],
      runtime: 97,
    },
  ],
  movie: {
    id: 412059,
    title: 'Mobile Homes',
    tagline: '',
    vote_average: 0,
    vote_count: 1,
    release_date: '2018-04-04',
    poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
    overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
    budget: 0,
    revenue: 0,
    genres: [
      'Drama',
    ],
    runtime: 105,
  },
  fetchMovie: () => ('Hello'),
  fetchSearchMovies: () => ('Hello'),
};

const store = mockStore({
  movies: [
    {
      id: 412059,
      title: 'Mobile Homes',
      tagline: '',
      vote_average: 0,
      vote_count: 1,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
      overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
      budget: 0,
      revenue: 0,
      genres: [
        'Drama',
      ],
      runtime: 105,
    },
    {
      id: 424585,
      title: 'Mademoiselle Paradis',
      tagline: '',
      vote_average: 6,
      vote_count: 2,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/k93b9hE30fdYG63atluVP5aDm2V.jpg',
      overview: '18th century Vienna. The true story of Maria Theresia von Paradis, a gifted piano player and close friend of Mozart’s, who lost her eye-sight as a child. Desperate to cure their talented daughter, the Paradis entrust Maria to Dr. Mesmer, a forward-thinking-physician who gives her the care and attention that she requires. With the doctor’s innovative techniques of magnetism, Maria slowly recovers her sight. But this miracle comes at a price as the woman progressively starts to lose her gift for music. Faced with a heavy dilemma, Mademoiselle Paradis will have to choose: an ordinary life in the light or an extraordinary life in darkness, as a virtuoso.',
      budget: 0,
      revenue: 0,
      genres: [
        'History',
        'Drama',
      ],
      runtime: 97,
    },
  ],
  movie: {
    id: 412059,
    title: 'Mobile Homes',
    tagline: '',
    vote_average: 0,
    vote_count: 1,
    release_date: '2018-04-04',
    poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
    overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
    budget: 0,
    revenue: 0,
    genres: [
      'Drama',
    ],
    runtime: 105,
  },
});

const failedData = {
  movies: [
    {
      id: 412059,
      title: 'Mobile Homes',
      tagline: '',
      vote_average: 0,
      vote_count: 1,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
      overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
      budget: 0,
      revenue: 0,
      genres: [
        'Drama',
      ],
      runtime: null,
    },
    {
      id: 424585,
      title: 'Mademoiselle Paradis',
      tagline: '',
      vote_average: 6,
      vote_count: 2,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/k93b9hE30fdYG63atluVP5aDm2V.jpg',
      overview: '18th century Vienna. The true story of Maria Theresia von Paradis, a gifted piano player and close friend of Mozart’s, who lost her eye-sight as a child. Desperate to cure their talented daughter, the Paradis entrust Maria to Dr. Mesmer, a forward-thinking-physician who gives her the care and attention that she requires. With the doctor’s innovative techniques of magnetism, Maria slowly recovers her sight. But this miracle comes at a price as the woman progressively starts to lose her gift for music. Faced with a heavy dilemma, Mademoiselle Paradis will have to choose: an ordinary life in the light or an extraordinary life in darkness, as a virtuoso.',
      budget: 0,
      revenue: 0,
      genres: [
        'History',
        'Drama',
      ],
      runtime: 97,
    },
  ],
  movie: {},
  fetchMovie: () => ('Hello'),
  fetchSearchMovies: () => ('Hello'),
};

const failedStore = mockStore({
  movies: [
    {
      id: 412059,
      title: 'Mobile Homes',
      tagline: '',
      vote_average: 0,
      vote_count: 1,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
      overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
      budget: 0,
      revenue: 0,
      genres: [
        'Drama',
      ],
      runtime: 105,
    },
    {
      id: 424585,
      title: 'Mademoiselle Paradis',
      tagline: '',
      vote_average: 6,
      vote_count: 2,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/k93b9hE30fdYG63atluVP5aDm2V.jpg',
      overview: '18th century Vienna. The true story of Maria Theresia von Paradis, a gifted piano player and close friend of Mozart’s, who lost her eye-sight as a child. Desperate to cure their talented daughter, the Paradis entrust Maria to Dr. Mesmer, a forward-thinking-physician who gives her the care and attention that she requires. With the doctor’s innovative techniques of magnetism, Maria slowly recovers her sight. But this miracle comes at a price as the woman progressively starts to lose her gift for music. Faced with a heavy dilemma, Mademoiselle Paradis will have to choose: an ordinary life in the light or an extraordinary life in darkness, as a virtuoso.',
      budget: 0,
      revenue: 0,
      genres: [
        'History',
        'Drama',
      ],
      runtime: 97,
    },
  ],
  movie: {},
});

const failedDataMovie = {
  movies: [
    {
      id: 412059,
      title: 'Mobile Homes',
      tagline: '',
      vote_average: 0,
      vote_count: 1,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
      overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
      budget: 0,
      revenue: 0,
      genres: [
        'Drama',
      ],
      runtime: null,
    },
    {
      id: 424585,
      title: 'Mademoiselle Paradis',
      tagline: '',
      vote_average: 6,
      vote_count: 2,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/k93b9hE30fdYG63atluVP5aDm2V.jpg',
      overview: '18th century Vienna. The true story of Maria Theresia von Paradis, a gifted piano player and close friend of Mozart’s, who lost her eye-sight as a child. Desperate to cure their talented daughter, the Paradis entrust Maria to Dr. Mesmer, a forward-thinking-physician who gives her the care and attention that she requires. With the doctor’s innovative techniques of magnetism, Maria slowly recovers her sight. But this miracle comes at a price as the woman progressively starts to lose her gift for music. Faced with a heavy dilemma, Mademoiselle Paradis will have to choose: an ordinary life in the light or an extraordinary life in darkness, as a virtuoso.',
      budget: 0,
      revenue: 0,
      genres: [
        'History',
        'Drama',
      ],
      runtime: 97,
    },
  ],
  fetchMovie: () => ('Hello'),
  fetchSearchMovies: () => ('Hello'),
};

const failedStoreMovie = mockStore({
  movies: [
    {
      id: 412059,
      title: 'Mobile Homes',
      tagline: '',
      vote_average: 0,
      vote_count: 1,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg',
      overview: 'In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.',
      budget: 0,
      revenue: 0,
      genres: [
        'Drama',
      ],
      runtime: 105,
    },
    {
      id: 424585,
      title: 'Mademoiselle Paradis',
      tagline: '',
      vote_average: 6,
      vote_count: 2,
      release_date: '2018-04-04',
      poster_path: 'https://image.tmdb.org/t/p/w500/k93b9hE30fdYG63atluVP5aDm2V.jpg',
      overview: '18th century Vienna. The true story of Maria Theresia von Paradis, a gifted piano player and close friend of Mozart’s, who lost her eye-sight as a child. Desperate to cure their talented daughter, the Paradis entrust Maria to Dr. Mesmer, a forward-thinking-physician who gives her the care and attention that she requires. With the doctor’s innovative techniques of magnetism, Maria slowly recovers her sight. But this miracle comes at a price as the woman progressively starts to lose her gift for music. Faced with a heavy dilemma, Mademoiselle Paradis will have to choose: an ordinary life in the light or an extraordinary life in darkness, as a virtuoso.',
      budget: 0,
      revenue: 0,
      genres: [
        'History',
        'Drama',
      ],
      runtime: 97,
    },
  ],
});

describe('MovieDetails', () => {
  it('renders correctly', () => {
    const component = TestRenderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <MovieDetails {...data} history={historyMock} location={{ pathname: '/movie/511679' }} />
        </MemoryRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const component = TestRenderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <MovieDetails {...data} history={historyMock} location={{ pathname: '/movie/511679' }} />
        </MemoryRouter>
      </Provider>,
    );
    component.root.findByProps({ className: 'details_button' }).props.onClick();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const component = TestRenderer.create(
      <Provider store={failedStore}>
        <MemoryRouter>
          <MovieDetails {...failedData} history={historyMock} location={{ pathname: '/movie/511679' }} />
        </MemoryRouter>
      </Provider>,
    );
    component.root.findByProps({ className: 'details_button' }).props.onClick();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const component = TestRenderer.create(
      <Provider store={failedStoreMovie}>
        <MemoryRouter>
          <MovieDetails {...failedDataMovie} history={historyMock} location={{ pathname: '/movie/511679' }} />
        </MemoryRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });

  test('lifecycle method should have been called', () => {
    const component = TestRenderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <MovieDetails {...data} history={historyMock} location={{ pathname: '/movie/511679' }} />
        </MemoryRouter>
      </Provider>,
    );
    component.unmount();
  });
});
