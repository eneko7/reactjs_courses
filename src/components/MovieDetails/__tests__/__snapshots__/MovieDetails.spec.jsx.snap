// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<div
  className="movie_details"
>
  <div
    className="top_line"
  >
    <span
      className="logo"
    >
      NetflixRoulette
    </span>
    <button
      className="details_button"
      onClick={[Function]}
      type="button"
    >
      search
    </button>
  </div>
  <div
    className="main"
  >
    <div
      className="movie_element_poster poster"
      style={
        Object {
          "backgroundImage": "linear-gradient(75deg, rgba(0, 0, 0, 1) 10%, rgba(255, 255, 255, 0) 30%), url(https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg)",
        }
      }
    />
    <div
      className="wrapper_right"
    >
      <span
        className="movie_titile"
      >
        Mobile Homes
      </span>
      <div
        className="genres genresClass"
      >
        <span
          className="genre genresClass"
        >
          Drama
        </span>
      </div>
      <div
        className="wrapper_year_and_time"
      >
        <span
          className="year"
        >
          2018
        </span>
        <span
          className="time"
        >
           105  min
        </span>
      </div>
      <div
        className="movie_description"
      >
        In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieDetails renders correctly 2`] = `
<div
  className="movie_details"
>
  <div
    className="top_line"
  >
    <span
      className="logo"
    >
      NetflixRoulette
    </span>
    <button
      className="details_button"
      onClick={[Function]}
      type="button"
    >
      search
    </button>
  </div>
  <div
    className="main"
  >
    <div
      className="movie_element_poster poster"
      style={
        Object {
          "backgroundImage": "linear-gradient(75deg, rgba(0, 0, 0, 1) 10%, rgba(255, 255, 255, 0) 30%), url(https://image.tmdb.org/t/p/w500/of72rtl3j0C9yJMlSJZHXnaByz8.jpg)",
        }
      }
    />
    <div
      className="wrapper_right"
    >
      <span
        className="movie_titile"
      >
        Mobile Homes
      </span>
      <div
        className="genres genresClass"
      >
        <span
          className="genre genresClass"
        >
          Drama
        </span>
      </div>
      <div
        className="wrapper_year_and_time"
      >
        <span
          className="year"
        >
          2018
        </span>
        <span
          className="time"
        >
           105  min
        </span>
      </div>
      <div
        className="movie_description"
      >
        In forgotten towns along the American border, a young mother drifts from one motel to the next with her intoxicating boyfriend and her 8-year-old son. The makeshift family scrapes by, living one hustle at a time, until the discovery of a mobile home community offers an alternative life.
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieDetails renders correctly 3`] = `
<div
  className="movie_details"
>
  <div
    className="top_line"
  >
    <span
      className="logo"
    >
      NetflixRoulette
    </span>
    <button
      className="details_button"
      onClick={[Function]}
      type="button"
    >
      search
    </button>
  </div>
  <div
    className="main"
  >
    <div
      className="movie_element_poster poster"
      style={
        Object {
          "backgroundImage": "linear-gradient(75deg, rgba(0, 0, 0, 1) 10%, rgba(255, 255, 255, 0) 30%), url()",
        }
      }
    />
    <div
      className="wrapper_right"
    >
      <span
        className="movie_titile"
      />
      <div
        className="wrapper_year_and_time"
      >
        <span
          className="year"
        >
          NaN
        </span>
        <span
          className="time"
        >
          
        </span>
      </div>
      <div
        className="movie_description"
      />
    </div>
  </div>
</div>
`;

exports[`MovieDetails renders correctly 4`] = `
<div
  className="loading"
>
  <div
    className="movies_grid_wrapper_loading"
    id="loading"
  >
    <div
      className="movies_grid_wrapper_loading"
    />
    <div
      className="movies_grid_wrapper_loading"
    />
    <div
      className="movies_grid_wrapper_loading"
    />
  </div>
  <div
    className="movies_grid_wrapper_loading_text"
    id="loading_text"
  >
    <span
      className="movies_grid_wrapper_loading_text_span"
    >
      LOADING
    </span>
  </div>
</div>
`;
